File: C++/CTCI/strings&array/prob1.cpp
#include <iostream>
#include <unordered_map>
#include <string>
using namespace std;

unordered_map <char,int> u_map;

int main()
{
  string a;
  bool flag = 0;
  cout<<"Enter the string: "<<endl;
  cin>>a;
  int n = a.length();
  cout<<n<<endl;
  char* i = &a[0];
  int ii=0;
  while(*i)
  {
    u_map[*i] = ++u_map[*i];
    cout<<"Char: "<<*i<<" num: "<<u_map[*i]<<endl;
    i++;
  }
  unordered_map <char, int> ::const_iterator itr;
  for (itr = u_map.begin();itr!=u_map.end(); itr++)
  {
    if (itr->second>1)
    {
      flag = 1; break;
    }
  }
  if(flag==1) cout<<"String is not Unique"<<endl;
  else cout<<"String is Unique"<<endl;
}

File: C++/CTCI/strings&array/prob1.py
string = raw_input('Enter a string: ')
i=0
flag = False
hashmap = {}
while i<len(string):
    hashmap[string[i]] =0
    i+=1
i=0
while i<len(string):
    hashmap[string[i]] +=1 
    i+=1
i=0
while i<len(string):
    if hashmap[string[i]]>1:
        flag =True
        break
    i+=1
if flag:
    print 'Not unique'
else:
    print "Unique"
